lib/_http_agent.js

```diff
@@ -409,14 +440,6 @@ function setRequestSocket(agent, req, socket) {
     return;
   }
   socket.setTimeout(req.timeout);
-  // Reset timeout after response end
-  req.once('response', (res) => {
-    res.once('end', () => {
-      if (socket.timeout !== agentTimeout) {
-        socket.setTimeout(agentTimeout);
-      }
-    });
-  });
```

Comment:
We only need to do this if re-using the socket. Move it to where it is put into the free list.